### PR TITLE
Docker patch

### PR DESCRIPTION
**Pull request by brtknr**

### CODE REVIEW

File: magnum/drivers/common/templates/swarm/fragments/add-docker-daemon-options.sh [hunk 1]
```diff
@@ -2,12 +2,42 @@
...
+    opts = {}
+
+opts["hosts"] = ["fd://"]
+if "${SET_LOG_DRIVER}" == "True":
+    opts["log-driver"] = "${LOG_DRIVER}"
+if "${SET_SELINUX_ENABLED}" == "True":
+    opts["selinux-enabled"] = True
+if "${SWARM_MODE}" == "True":
+    opts["hosts"].append("tcp://0.0.0.0:2376")
+else:
```
Comment (markgoddard) [inline magnum/drivers/common/templates/swarm/fragments/add-docker-daemon-options.sh]: The swarm mode stuff I added is no longer in use - we can remove L31-33, and deindent L34.

File: magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh [hunk 2]
```diff
@@ -11,13 +16,40 @@ clear_docker_storage () {
     # clear storage graph
```
Comment (markgoddard) [inline magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh]: I think L15 needs to move to L28 - it does not apply for no dss.

File: magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh [hunk 3]
```diff
@@ -11,13 +16,40 @@ clear_docker_storage () {
...
+    if has_dss; then
+        clear_docker_storage_dss
+    else
+        clear_docker_storage_no_dss
+    fi
+}
+
+clear_docker_storage_dss () {
+    # remove current LVs
+    docker-storage-setup --reset
```
Comment (markgoddard) [inline magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh]: This line was replaced with `systemctl disable docker-storage-setup` upstream (see 28fff8006ae17d9afeee4ad409e12b6608fa26cb).

File: magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh [hunk 4]
```diff
@@ -29,10 +61,36 @@ configure_storage_driver_generic() {
     echo "DOCKER_STORAGE_OPTIONS=\"--storage-driver $1\"" > /etc/sysconfig/docker-storage
 }
 
+configure_storage_driver_generic_no_dss () {
+    # Configure dockerd to use the devicemapper volume.
```
Comment (markgoddard) [inline magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh]: Comment needs updating

**Comment by brtknr** [comment]
Thanks for the review @markgoddard, I've addressed your comments in the new commit.

File: magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh [hunk 5]
```diff
@@ -1,23 +1,54 @@
...
+}
+
 # * Remove any existing docker-storage configuration. In case of an
 #   existing configuration, docker-storage-setup will fail.
 # * Remove docker storage graph
 clear_docker_storage () {
     # stop docker
     systemctl stop docker
     systemctl disable docker-storage-setup
-    # clear storage graph
```
Comment (markgoddard) [inline magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh]: The comment is below the line being commented on - the line that should have moved is L15 (systemctl disable docker-storage-setup).

Comment (markgoddard) [reply magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh]: My bad, I think I commented on the wrong line.

File: magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh [hunk 6]
```diff
@@ -26,7 +26,6 @@ clear_docker_storage_dss () {
     # remove current LVs
     systemctl disable docker-storage-setup
 
-    # clear storage graph
```
Comment (brtknr) [inline magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh]: @markgoddard is this what you meant?

**Pull request merged**